framework/lib/components/text-field/formatted-number-input-field.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { identity } from 'ramda'
+import { FormattedNumberInputFieldProps } from './types'
+import { Field } from '../form'
+import { FormattedNumberInput } from './formatted-number-input'
+
+/**
+ * The formatted number input component wrapped in a <Field />
+ * meant to be used only inside <Form />
+ * @see FormattedNumberInput
+ * @see {@link https://northlight.dev/reference/formatted-number-input-field}
+ * @example (Example)
+ * (?
+ * <Form initialValues={{money: ''}}>
+ *  <FormattedNumberInputField
+ *   name="money"
+ *   label="How much money is currently on your account"
+ *   labelPosition="left"
+ *  />
+ * </Form>
+ * ?)
+ *
+ */
+export const FormattedNumberInputField = ({
+  name,
+  label,
+  isRequired,
+  validate,
+  onChange: onChangeCallback = identity,
+  direction = 'row',
+  ...rest
+}: FormattedNumberInputFieldProps) => (
+  <Field
+    name={ name }
+    label={ label }
+    isRequired={ isRequired }
+    direction={ direction }
+    validate={ validate }
+  >
+    { ({ onChange, value }) => (
+      <FormattedNumberInput
+        name={ name }
+        onChange={ (values, sourceInfo) => {
+          onChange(values.floatValue)
+          onChangeCallback(values, sourceInfo)
+        } }
+        value={ value }
+        data-testid="formatted-number-input-field-id"
```

Comment:
shouldnt we send in test ids from consumer. so they can select elements more granurally if we render multiple in one screen?

Reply:
yes